components/signatures/permit/PermitsEntry.tsx

```diff
@@ -24,6 +25,8 @@ const PermitsEntry = ({ token }: Props) => {
   const { address, selectedChainId } = useAddressPageContext();
   const handleTransaction = useHandleTransaction();
 
+  const isPreviouslyCancelled = filterLastCancelled(useAddressEvents().events, token).alreadyCancelled;
```

Comment:
I really like this addition to prevent people from needlessly cancelling signatures multiple times. But instead of checking if the *ever* cancelled this signature, we should probably only check the past hour or something (so compare the last cancelled time with the current time).